### PR TITLE
Update copyRequest code.

### PR DESCRIPTION
**Pull request by mdwn**
The copyRequest code has been adapted from upstream. The `URL.Opaque` assignment was inadvertently causing double question marks to appear when a request URL is terminated with a question mark. That is:

```
https://www.example.com?
```

would show up as

```
https://www.example.com??
```

By avoiding the assignment to `URL.Opaque`, this no longer occurs.

Note: nodered appears to run just fine with this change in place.

Addresses https://github.com/gravitational/teleport/issues/17460

### CODE REVIEW

File: forward/fwd.go [hunk 1]
```diff
@@ -206,22 +206,39 @@ func (f *httpForwarder) serveHTTP(w http.ResponseWriter, req *http.Request, ctx
...
+	outReq.URL = utils.CopyURL(outReq.URL)
+	outReq.URL.Scheme = target.Scheme
+	outReq.URL.Host = target.Host
+
+	u := f.getURLFromRequest(outReq, ctx)
+
+	outReq.URL.Path = u.Path
+	outReq.URL.RawPath = u.RawPath
+	outReq.URL.RawQuery = u.RawQuery
+	outReq.RequestURI = "" // Outgoing request should not have RequestURI
```
Comment (r0mant) [inline forward/fwd.go:240]: So as far as I understand, the actual fix is instead of setting the entire request URI as "opaque" to parse the URL and set parts like path/query individually, correct?

If so, what does the `getURLFromRequest` change fix? Which scenario we could be using incorrect URL in? Just want to make sure we understand why this change is needed so we don't break anything.

Comment (mdwn) [reply forward/fwd.go:240]: FWIW none of this is actually my code save for massaging it into our fork -- this is all from upstream. I'm unsure what the `getURLFromRequest` call was originally introduced for, but it sounds like it's a way of ensuring the URL is retrieved properly from the request regardless of how it's created, either through programmatic creation of an `http.Request` or sending the request through an actual http client.

Comment (mdwn) [reply forward/fwd.go:240]: https://github.com/vulcand/oxy/pull/108 Is the relevant PR with sadly very little description.

Comment (mdwn) [reply forward/fwd.go:240]: @r0mant And I just realized I didn't answer your first question. Yes, that's correct, not using `Opaque` and doing an explicit parse fixes this.

Comment (r0mant) [reply forward/fwd.go:240]: Looking at the linked PR and the comment, seems like this case was specifically for the test which creates `http.NewRequest` itself so we shouldn't really hit it in production.

**Review by r0mant — APPROVED** [review]
lgtm but can you please test it with some apps other than Nodered too, e.g. Grafana.

**Comment by mdwn** [comment]
Can confirm this worked well with Grafana, Nodered, the Teleport Websockets app, and Jupyter.

**Comment by greedy52** [comment]
This seems broke AWS access for ex `tsh aws s3 cp file s3://some-bucket/`

https://github.com/gravitational/teleport/blob/42dde44e1b2cacbe924d97c20e92163ea7bbe265/lib/srv/app/aws/handler.go#L202-L213

But i am still trying to find out why

**Comment by mdwn** [comment]
> This seems broke AWS access for ex `tsh aws s3 cp file s3://some-bucket/`
> 
> https://github.com/gravitational/teleport/blob/42dde44e1b2cacbe924d97c20e92163ea7bbe265/lib/srv/app/aws/handler.go#L202-L213
> 
> But i am still trying to find out why

I'm guessing because `Opaque` is no longer used by oxy here (this is true of upstream), so I'm guessing it's not set?

**Comment by greedy52** [comment]
> > This seems broke AWS access for ex `tsh aws s3 cp file s3://some-bucket/`
> > https://github.com/gravitational/teleport/blob/42dde44e1b2cacbe924d97c20e92163ea7bbe265/lib/srv/app/aws/handler.go#L202-L213
> > But i am still trying to find out why
> 
> I'm guessing because `Opaque` is no longer used by oxy here (this is true of upstream), so I'm guessing it's not set?

ya. URL.Opaque is empty now, missing path.

should be an easy fix on the aws app side. will do tmr
